src/EnergyPlus/api/datatransfer.cc

```diff
@@ -116,7 +116,8 @@ char *listAllAPIDataCSV(EnergyPlusState state)
     // note that we cannot just return a c_str to the local string, as the string will be destructed upon leaving
     // this function, and undefined behavior will occur.
     // instead make a deep copy, and the user must manage the new char * pointer
-    char *p = new char[std::strlen(output.c_str())];
+    // strcpy copies including the null-terminator, strlen doesn't include it
+    char *p = new char[std::strlen(output.c_str()) + 1];
```

Comment:
Here you go

Reply:
@jmarrec is this just draft because you haven't got a test for it yet?  Or is there additional work to do?  Thanks!

Reply:
I was just quick fixing (a unit test for this would be way overkilled), and I think my default github settings lately are open as draft, I didn't realize this was marked as draft.